executor/set.go

```diff
@@ -115,6 +115,14 @@ func (e *SetExecutor) setSysVariable(ctx context.Context, name string, v *expres
 		}
 		return variable.ErrUnknownSystemVar.GenWithStackByArgs(name)
 	}
+
+	if sysVar.HasInstanceScope() && !v.IsGlobal && sessionVars.EnableLegacyInstanceScope {
+		// For backward compatibility we will change the v.IsGlobal to true,
+		// and append a warning saying this will not be supported in future.
+		v.IsGlobal = true
+		sessionVars.StmtCtx.AppendWarning(errors.New("allowing scope change for now, but its deprecated"))
```

Comment:
For this error, we should reserve a new TiDB error code. This will mean it shows up uniquely in information_schema here: https://docs.pingcap.com/tidb/stable/client-errors-summary-global#client_errors_summary_global

To add a new error:
- Edit `errno/errcode.go` (pick the next 81xx number) and `errno/errname.go`
- Add the error to the executor package at `executor/errors.go`.

We also need to add a test for this behavior where setting session scope "switches" to global scope if EnableLegacyInstanceScope is set.